src/main/kotlin/dev/znci/rocket/scripting/globals/tables/AABB.kt

```diff
@@ -0,0 +1,182 @@
+    }
+
+
+    @TwineNativeFunction
+    fun expand(amount: Double): LuaAABB {
+        bukkitBox.expand(amount)
+        return this
+    }
+
+    @TwineNativeFunction
+    @TwineOverload
+    fun expand(x: Double, y: Double, z: Double): LuaAABB {
+        bukkitBox.expand(x, y, z)
+        return this
+    }
+
+    @TwineNativeFunction
+    @TwineOverload
+    fun expand(direction: LuaVector3, amount: Double): LuaAABB {
+        bukkitBox.expand(direction.xProperty, direction.yProperty, direction.zProperty, amount)
+        return this
+    }
+
+    @TwineNativeFunction
+    fun union(other: LuaAABB): LuaAABB {
+        bukkitBox.union(other.bukkitBox)
+        return this
+    }
+
+    @TwineNativeFunction
+    fun expandToContainAABB(other: LuaAABB): LuaAABB {
+        return union(other)
+    }
+
+    @TwineNativeFunction
+    fun shift(x: Double, y: Double, z: Double): LuaAABB {
+        bukkitBox.shift(x, y, z)
+        return this
+    }
+
+    @TwineNativeFunction
+    @TwineOverload
+    fun shift(offset: LuaVector3): LuaAABB {
+        bukkitBox.shift(offset.xProperty, offset.yProperty, offset.zProperty)
+        return this
+    }
+
+    @TwineNativeFunction
+    fun rayTrace(start: LuaVector3, direction: LuaVector3, maxDistance: Double): LuaVector3? {
+        val dir = direction.clone().normalize()
+        val result = bukkitBox.rayTrace(
+            start.toBukkitVector(),
+            dir.toBukkitVector(),
+            maxDistance
+        ) ?: return null
+        return result.hitPosition.toLuaVector3()
+    }
+
+    @TwineNativeFunction
+    fun intersection(other: LuaAABB): LuaAABB {
+        return LuaAABB(bukkitBox.intersection(other.bukkitBox))
+    }
+
+    @TwineNativeFunction
+    fun closestPointTo(point: LuaVector3): LuaVector3 {
+        return LuaVector3(
+            point.xProperty.coerceIn(bukkitBox.minX, bukkitBox.maxX),
+            point.yProperty.coerceIn(bukkitBox.minY, bukkitBox.maxY),
+            point.zProperty.coerceIn(bukkitBox.minZ, bukkitBox.maxZ)
+        )
+    }
+
+    @TwineNativeFunction
+    fun distanceTo(point: LuaVector3): Double {
+        return closestPointTo(point).distance(point)
+    }
+
+    @TwineNativeProperty("center")
+    val center: LuaVector3
+        get() = LuaVector3(
+            bukkitBox.centerX,
+            bukkitBox.centerY,
+            bukkitBox.centerZ
+        )
+
+    @TwineNativeProperty("volume")
+    val volume: Double
+        get() = bukkitBox.volume
+
+    @TwineNativeProperty("size")
+    val size: LuaVector3
+        get() = LuaVector3(bukkitBox.widthX,  bukkitBox.height, bukkitBox.widthZ)
+
+    @TwineNativeProperty("min")
+    val minProperty: LuaVector3
+        get() = bukkitBox.min.toLuaVector3()
+
+    @TwineNativeProperty("max")
+    val maxProperty: LuaVector3
+        get() = maxProperty
+
+    @TwineNativeFunction
+    override fun toString(): String {
+        return "AABB($minProperty, $maxProperty)"
```

Comment:
styling issue, should reflect the name of the class, 
`class LuaAABBs : TwineNative("aabb") {`
should then be `return "AABB($minProperty, $maxProperty)"`
not a major issue though